components/feature/downloads/src/test/java/mozilla/components/feature/downloads/share/ShareDownloadFeatureTest.kt

```diff
@@ -0,0 +1,281 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.downloads.share
+
+import android.content.Context
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.TestCoroutineDispatcher
+import kotlinx.coroutines.test.TestCoroutineScope
+import mozilla.components.browser.state.action.ShareInternetResourceAction
+import mozilla.components.browser.state.state.BrowserState
+import mozilla.components.browser.state.state.ContentState
+import mozilla.components.browser.state.state.TabSessionState
+import mozilla.components.browser.state.state.content.ShareInternetResourceState
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.concept.fetch.Client
+import mozilla.components.concept.fetch.MutableHeaders
+import mozilla.components.concept.fetch.Request
+import mozilla.components.concept.fetch.Response
+import mozilla.components.support.test.any
+import mozilla.components.support.test.argumentCaptor
+import mozilla.components.support.test.ext.joinBlocking
+import mozilla.components.support.test.mock
+import mozilla.components.support.test.robolectric.testContext
+import mozilla.components.support.test.rule.MainCoroutineRule
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.doAnswer
+import org.mockito.Mockito.doNothing
+import org.mockito.Mockito.doReturn
+import org.mockito.Mockito.spy
+import org.mockito.Mockito.verify
+import java.io.File
+import java.nio.charset.StandardCharsets
+
+@RunWith(AndroidJUnit4::class)
+class ShareDownloadFeatureTest {
+    private val testDispatcher = TestCoroutineDispatcher()
+
+    @get:Rule
+    val coroutinesTestRule = MainCoroutineRule(testDispatcher)
+
+    @Test
+    fun `cleanupCache should automatically be called when this class is initialized`() = runBlocking {
+        val context: Context = mock()
+        val cacheDir = File(context.cacheDir, cacherDirName).also { dir ->
+            dir.mkdirs()
+            File(dir, "leftoverFile").also { file ->
+                file.createNewFile()
+            }
+        }
+
+        assertTrue(cacheDir.listFiles()!!.isNotEmpty())
+
+        ShareDownloadFeature(context, mock(), mock(), null)
+        // Production code uses launch to start the cleanup. Let's give it time to complete
+        delay(100)
```

Comment:
nit: This could make the test flaky, better to replace the `CoroutineScope(IO)` in test by main scope.

Reply:
Thanks!
Added a top level var for the coroutine context that can be changed when testing even before the object being created so that it will be used in init.